internal/reconciliations/external_authorizer/external_authorizer_test.go

```diff
@@ -0,0 +1,240 @@
+					Config: operatorv1alpha2.Config{
+						Authorizers: []*operatorv1alpha2.Authorizer{
+							{
+								Name:    expectedName,
+								Service: expectedService,
+								Port:    expectedPort,
+							},
+							{
+								Name:    expectedName,
+								Service: expectedService,
+								Port:    expectedPort,
+							},
+						},
+					},
+				},
+			}
+
+			//when
+			err := reconciler.Reconcile(context.TODO(), istioCR)
+
+			//then
+			Expect(err).To(HaveOccurred())
+			Expect(err.Description()).To(Equal(fmt.Sprintf("Authorizer name needs to be unique: %s is duplicated", expectedName)))
+		})
+	})
+
+	Context("ServiceEntry creation", func() {
+
+		It("should succeed creating ServiceEntry", func() {
+			//given
+			client := createFakeClient()
+			reconciler := external_authorizer.NewReconciler(client)
+
+			//when
+			err := reconciler.Reconcile(context.TODO(), istioCR)
+
+			//then
+			Expect(err).To(Not(HaveOccurred()))
+
+			var s networkingv1alpha3.ServiceEntryList
+			listErr := client.List(context.TODO(), &s)
+			Expect(listErr).To(Not(HaveOccurred()))
+			Expect(s.Items).To(HaveLen(1))
+		})
+
+		It("should succeed configuring ServiceEntry", func() {
+			//given
+			client := createFakeClient()
+			reconciler := external_authorizer.NewReconciler(client)
+
+			//when
+			err := reconciler.Reconcile(context.TODO(), istioCR)
+
+			//then
+			Expect(err).To(Not(HaveOccurred()))
+
+			var s networkingv1alpha3.ServiceEntryList
+			listErr := client.List(context.TODO(), &s)
+			Expect(listErr).To(Not(HaveOccurred()))
+			Expect(s.Items).To(HaveLen(1))
+
+			Expect(s.Items[0].Name).To(Equal(expectedName))
+
+			Expect(s.Items[0].Spec.Ports).To(HaveLen(1))
+			Expect(s.Items[0].Spec.Ports[0].Protocol).To(Equal("http"))
+			Expect(s.Items[0].Spec.Ports[0].Name).To(Equal("http"))
+			Expect(s.Items[0].Spec.Ports[0].Number).To(BeEquivalentTo(expectedPort))
+
+			Expect(s.Items[0].Spec.Hosts).To(HaveLen(1))
+			Expect(s.Items[0].Spec.Hosts[0]).To(Equal(expectedService))
+		})
+
+	})
+
+	Context("ServiceEntry cleanup", func() {
+
+		It("should remove ServiceEntry when authorizer is removed from Istio CR", func() {
+			//given
+			client := createFakeClient()
+			reconciler := external_authorizer.NewReconciler(client)
+
+			//when
+			err := reconciler.Reconcile(context.TODO(), istioCR)
+
+			//then
+			Expect(err).To(Not(HaveOccurred()))
+
+			var s networkingv1alpha3.ServiceEntryList
+			listErr := client.List(context.TODO(), &s)
+			Expect(listErr).To(Not(HaveOccurred()))
+			Expect(s.Items).To(HaveLen(1))
+
+			// given
+			updatedIstioCr := istioCR.DeepCopy()
+			updatedIstioCr.Spec.Config = operatorv1alpha2.Config{}
+
+			// when
+			err = reconciler.Reconcile(context.TODO(), *updatedIstioCr)
+
+			// then
+			Expect(err).To(Not(HaveOccurred()))
+
+			listErr = client.List(context.TODO(), &s)
+			Expect(listErr).To(Not(HaveOccurred()))
+			Expect(s.Items).To(HaveLen(0))
+		})
+
+		It("should not owned service entries", func() {
```

Comment:
The test description might be a little bit confusing.
```suggestion
		It("should not remove service entries not owned by the module", func() {
```